File: CS/DataStructure/B10845-S4.cs
using System.Text;

class B10845_S4
{
  static void Main()
  {
    int a = int.Parse(Console.ReadLine());
    StringBuilder sb = new StringBuilder();
    Queue<int> q = new Queue<int>();
    int value = -1;
    for (int i = 0; i < a; i++)
    {
      string[] s = Console.ReadLine().Split();
      string op = s[0];
      switch (op)
      {
        case "push":
          value = int.Parse(s[1]);
          q.Enqueue(value);
          break;
        case "pop":
          sb.AppendLine(q.Count == 0 ? "-1" : q.Dequeue().ToString());
          break;
        case "size":
          sb.AppendLine(q.Count.ToString());
          break;
        case "empty":
          sb.AppendLine(q.Count == 0 ? "1" : "0");
          break;
        case "front":
          sb.AppendLine(q.Count == 0 ? "-1" : q.Peek().ToString());
          break;
        case "back":
          sb.AppendLine(q.Count == 0? "-1" : value.ToString());
          break;
      }
    }
    Console.Write(sb.ToString());
  }
}

File: py/algorithm/data_structure/b10845_s4.py
import sys
input = sys.stdin.readline
N = int(input())
q = list()
for i in range(N):
    s = input().split()
    op = s[0]
    if op == 'push':
        q.append(s[1])
    elif op == 'pop':
        print(q.pop(0) if q else -1)  # 큐는 선입선출이므로 pop(0)으로 첫 번째 요소를 제거. pop()은 리스트의 마지막을 제거함.
    elif op == 'size':
        print(len(q))
    elif op == 'empty':
        print(0 if q else 1)
    elif op == 'front':
        print(q[0] if q else -1)
    elif op == 'back':
        print(q[-1] if q else -1)
